### ISSUE TITLE
Default merge function for pair and triple should handle easy cases first

### ISSUE THREAD

**Issue by kayceesrk** [mirage/irmin#416]
The default merge functions for [pair](https://github.com/mirage/irmin/blob/master/src/ir_merge.ml#L137) and [triple](https://github.com/mirage/irmin/blob/master/src/ir_merge.ml#L149) go for the recursive case without attempting the easy cases. The easy cases are for example,

```ocaml
let pair (da,a) (db,b) =
  let dt = Ir_type.pair da db in
  let pp = Ir_type.dump dt in
  dt, fun ~old x y -> (* assuming non-monadic values for simplicity *)
    if x = y then x
    else if x = old then y
    else if y = old then x
    else (* recursive case *)
```

**Comment by samoht** [comment]
The fix for this is to wrap the combinator with the `default` merge function. This is a regression from the `0.*` releases.

**Comment by kayceesrk** [comment]
See #420.

**Comment by kayceesrk** [comment]
Fixed by #420. Closing issue.
